statistics-dashboard-service/src/main/java/eu/europeana/statistics/dashboard/service/StatisticsService.java

```diff
@@ -92,7 +95,15 @@ public FilteringResult queryDataWithFilters(StatisticsFilteringRequest statistic
     Map<MongoStatisticsField, ValueRange> rangeAvailableOptions = prepareRangeFilteringOptionsQuery(readyQuery);
 
     int totalRecords = queryResult.getRecordCount();
-    StatisticsResult statisticsAllRecordsResult = new StatisticsResult(STATISTICS_RESULT_ROOT_VALUE, totalRecords, 100);
+
+    StatisticsResult statisticsAllRecordsResult;
+    if(totalRecords == 0) {
```

Comment:
I think that the percentage could be "calculated" in the constructor(and setCount).
I mean it is based on a the total record number, it's not a value that we get from somewhere and it's relevant to that specific class.
So I would move that responsibility to the class.
The percentage would not be part of the constructor parameters, nor being allowed to be set(setter) and only allow it to be calculated inside the class (constructor/setCount).
How does that sound?